add windows on arm builds

DioEgizio opened the pull request:
This adds Windows on ARM builds. Note that while builds work natively on woa, we only have support for native windows on arm java on 1.19+.

Also includes a refactor to not use ecmqueryqt on qt6 since it broke on windows on arm and it's not needed there anyways

Also drops Windows-MinGW-w64-Legacy to bring back the number of builds as before the pr and windows 7/8.1 users can just use Windows-MSVC-Legacy

DioEgizio committed:
feat(actions) add arm64 build

Signed-off-by: DioEgizio <83089242+DioEgizio@users.noreply.github.com>

TheLastRar committed:
Use ECMQueryQt Only with Qt5

Qt6 provides CMake variables for the paths we need

Signed-off-by: TheLastRar <TheLastRar@users.noreply.github.com>

TheLastRar committed:
Remove unused QT_DATA_DIR/QT_MKSPECS_DIR

Signed-off-by: TheLastRar <TheLastRar@users.noreply.github.com>

TheLastRar committed:
Maybe fix ResourceFolderModel Clang64 test failure

Previously this test would fail sometimes without including ECMQueryQt

Signed-off-by: TheLastRar <TheLastRar@users.noreply.github.com>

DioEgizio committed:
refactor+fix: refactor the windows part of the release workflow and add arm to it

Signed-off-by: DioEgizio <83089242+DioEgizio@users.noreply.github.com>

ryanccn reviewed (approved):
I approved but it doesn't count :trollface:

And yes it works, I've tested it

Scrumplex commented on CMakeLists.txt:
```diff
@@ -235,12 +235,16 @@ else()
+    ecm_query_qt(QT_LIBS_DIR     QT_INSTALL_LIBS)
+    ecm_query_qt(QT_LIBEXECS_DIR QT_INSTALL_LIBEXECS)
+else()
+    set(QT_PLUGINS_DIR  ${QT${QT_VERSION_MAJOR}_INSTALL_PREFIX}/${QT${QT_VERSION_MAJOR}_INSTALL_PLUGINS})
+    set(QT_LIBS_DIR     ${QT${QT_VERSION_MAJOR}_INSTALL_PREFIX}/${QT${QT_VERSION_MAJOR}_INSTALL_LIBS})
+    set(QT_LIBEXECS_DIR ${QT${QT_VERSION_MAJOR}_INSTALL_PREFIX}/${QT${QT_VERSION_MAJOR}_INSTALL_LIBEXECS})
```
What's up with this? ECMQueryQt supports Qt 6 paths.
I think this makes the code less readable

DioEgizio replied:
it breaks on windows on arm cross compilation (uses host Qt paths instead of native ones)

TheLastRar replied:
The issue occurs with cross compiling for Windows arm, Qt uses both an install for the Host architecture (x64) and for the target/install architecture (Arm64)

ECMQueryQt however seems to pick qtpaths in the host Qt directory instead of the one in the Install Qt directory, this results in you getting paths for applicable host Qt, resulting in x64 Qt dlls in the arm64 PrismLauncher bundles. (This issue occurs with both the official Qt installs and the aqtinstalls

Additionally, the msvc arm64 qtpaths is actually a batch script to launch qtpaths from host Qt with a extra config (to give correct install paths), which aqtinstall doesn't setup correctly.

Scrumplex replied:
In this case you could remove `cmake/ECMQueryQt.cmake`, as it was only included in-tree for Qt 6 support

DioEgizio replied:
done

DioEgizio replied:
seems to fail without it :/

DioEgizio replied:
seems like ubuntu 20.04 uses ecm 5.92 but that was added in 5.93, gonna revert

flowln reviewed (approved):
more builds :iea:

DioEgizio committed:
chore: drop Windows-MinGW-w64-Legacy

this brings back the number of builds as the same as before the pr

this is one of the slowest builds + windows 7/8.1 users can just use Windows-MSVC-Legacy

Signed-off-by: DioEgizio <83089242+DioEgizio@users.noreply.github.com>
